src/hasAtLeast.ts

```diff
@@ -1,14 +1,41 @@
-import type { ReadonlyTuple } from "./_types";
+import type { IterableContainer } from "./_types";
 import { purry } from "./purry";
 
-type ArrayMinN<T, N extends number> = number extends N
-  ? // We can only compute the type for a literal number!
-    Array<T>
-  : // I don't know why we need to special-case the 0 case, but otherwise
-    // typescript complains we have a deep recursion.
-    N extends 0
-    ? Array<T>
-    : [...ReadonlyTuple<T, N>, ...Array<T>];
+type ArraySetRequired<
+  T extends IterableContainer,
+  N extends number,
+  Iteration extends ReadonlyArray<unknown> = [],
```

Comment:
**nit:** maybe `Result`?

Reply:
Oops, I was supposed to add `undefined` to this type, not the Head. Notice that we never return this type, we return T from within the recursion. This is just the regular typescript trick to stop recursions after N iterations.